### PR TITLE
portal: Stub out Settings views

### PR DESCRIPTION
**Pull request by jamilbk**
Adds Setting UI views based on the Balsamiq Wireframes. This should be merged **after** #1679 
Screenshot 2023-06-26 at 4 48 55 PM

Screenshot 2023-06-26 at 4 49 01 PM
Screenshot 2023-06-29 at 12 29 26 AM

Screenshot 2023-06-29 at 12 29 14 AM

Screenshot 2023-06-29 at 2 36 43 PM
Screenshot 2023-06-29 at 2 36 56 PM
Screenshot 2023-06-29 at 2 37 05 PM

### CODE REVIEW

**Comment by jamilbk** [comment]
@bmanifold There are still the API views and the Edit Identity Provider view, but I'll leave those as an exercise for the reader in order to move on, but they shouldn't comprise anything that hasn't already been stubbed out elsewhere.

File: elixir/apps/web/assets/js/event_listeners.js
Comment (jamilbk) [inline elixir/apps/web/assets/js/event_listeners.js:1]: See https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.JS.html#dispatch/2-examples

File: elixir/apps/web/lib/web/components/form_components.ex
Comment (jamilbk) [inline elixir/apps/web/lib/web/components/form_components.ex]: @AndrewDryga

Does this UI accomplish what you're planning for the backend? See screenshots in the issue description for more detail. I decided to decouple the provisioning strategy from the authentication method since some providers support SCIM with both OIDC and SAML (Azure, Authentik).

**Comment by jamilbk** [comment]
Looks like CI is failing due to a new bug in Elixir: https://github.com/elixir-lang/elixir/issues/12691

File: elixir/apps/web/lib/web/components/form_components.ex
Comment (AndrewDryga) [reply elixir/apps/web/lib/web/components/form_components.ex]: @jamilbk I did not plan to implement SCIM for OIDC (since every provider that supports that also supports SAML), but the general structure looks good.

File: elixir/apps/web/lib/web/components/form_components.ex
Comment (AndrewDryga) [inline elixir/apps/web/lib/web/components/form_components.ex]: I think we should not keep components like this one here, shared components are for things used everywhere in the app (eg. we should move `input`, `simple_form`, and other related components here) while this one is specific for identity provider views. Need to think about a better place to keep reusable things like this one without polluting the global namespace.

File: elixir/apps/web/lib/web/components/form_components.ex
Comment (jamilbk) [reply elixir/apps/web/lib/web/components/form_components.ex]: @AndrewDryga Gotcha -- it could be possible that our SCIM implementation doesn't require much, if any work to be auth method agnostic since it's just a REST API essentially. I think if the resource IDs match (e.g. `sub` in OIDC and `user.id` in SAML), it should theoretically work for both types of providers.

File: elixir/apps/web/lib/web/components/form_components.ex
Comment (jamilbk) [reply elixir/apps/web/lib/web/components/form_components.ex]: Good catch, yeah, I'll move this somewhere else.

File: elixir/apps/web/lib/web/components/form_components.ex
Comment (jamilbk) [reply elixir/apps/web/lib/web/components/form_components.ex]: If it turns out to come to us easy, it might be a good idea for SCIM to be auth method agnostic because of that pesky admin that wants SCIM with their OIDC connector in Azure.

File: elixir/apps/web/lib/web/components/form_components.ex
Comment (jamilbk) [reply elixir/apps/web/lib/web/components/form_components.ex]: Updated to keep these in a local `components.ex` file relative to the other views that depend on them.

File: elixir/apps/web/lib/web/components/form_components.ex
Comment (jamilbk) [reply elixir/apps/web/lib/web/components/form_components.ex]: The more trickier part would be implementing JIT provisioning for SAML (parsing XML doesn't seem fun), but maybe we can just disable that for now.

Comment (jamilbk) [reply elixir/apps/web/lib/web/components/form_components.ex]: Followed up over DM. SCIM is going to be tedious. Will approach it in stages and disable SCIM option until ready.

**Review by jamilbk — COMMENTED** [review]
Just waiting on approval from @bmanifold or @AndrewDryga

**Review by bmanifold — APPROVED** [review]
LGTM

**Comment by jamilbk** [comment]
Looks like @AndrewDryga needs to check off as well.

**Review by AndrewDryga — APPROVED** [review]

**Pull request merged**
